inference-engine/src/vpu/common/src/ngraph/transformations/merge_gather_gather_elements.cpp

```diff
@@ -0,0 +1,145 @@
+            const auto& m_squeezeAxes = patternMap.at(squeezeAxes);
+            const auto& m_squeeze = patternMap.at(squeeze);
+            const auto& m_gatherData = patternMap.at(gatherData);
+            const auto& m_gatherIndices = patternMap.at(gatherIndices);
+            const auto& m_gather = patternMap.at(gather);
+
+            for (const auto& gatherElement : gatherElements) {
+                const auto transposeIndices = std::make_shared<ngraph::opset6::Transpose>(gatherElement->input_value(1), m_transposePerm);
+                const auto unsqueezeIndices = std::make_shared<ngraph::opset6::Unsqueeze>(
+                    transposeIndices,
+                    m_squeezeAxes);
+                const auto expGatherElements = std::make_shared<ngraph::vpu::op::ExpGatherElements>(
+                    m_gatherData,
+                    unsqueezeIndices,
+                    m_gatherIndices,
+                    ngraph::as_type<ngraph::opset6::GatherElements>(gatherElement)->get_axis(),
+                    ngraph::as_type<ngraph::opset6::Gather>(m_gather.get_node())->get_axis());
+                const auto squeezeData = m_squeeze.get_node()->clone_with_new_inputs({expGatherElements, m_squeezeAxes});
+                const auto transposeData = m_transpose.get_node()->clone_with_new_inputs({squeezeData, m_transposePerm});
+                transposeData->set_friendly_name(gatherElement->get_friendly_name());
+                gatherElement->get_default_output().replace(transposeData);
+                wasGraphChanged = true;
+            }
+
+            for (const auto& shapeOf : shapeOfs) {
+                const auto gatherDataShape = std::make_shared<ngraph::opset6::ShapeOf>(
+                    m_gatherData,
+                    ngraph::as_type<ngraph::opset6::ShapeOf>(shapeOf)->get_output_type());
+                const auto gatherIndicesShape = std::make_shared<ngraph::opset6::ShapeOf>(
+                    m_gatherIndices,
+                    ngraph::as_type<ngraph::opset6::ShapeOf>(shapeOf)->get_output_type());
+                const auto axis = ngraph::as_type<ngraph::opset6::Gather>(m_gather.get_node())->get_axis();
+
+                const auto gatherIndicesRank = m_gatherIndices.get_partial_shape().rank().get_length();
+                const auto gatherDataRank = m_gatherData.get_partial_shape().rank().get_length();
+                const auto gatherOutRank = gatherDataRank + gatherIndicesRank - 1;
+
+                std::vector<int64_t> squeezeOutIndices(gatherOutRank);
+                std::iota(squeezeOutIndices.begin(), squeezeOutIndices.end(), 0);
+
+                auto squeezeAxes = ngraph::as_type<ngraph::opset6::Constant>(m_squeezeAxes.get_node())->cast_vector<int64_t>();
+                ngraph::normalize_axes(m_squeezeAxes.get_node()->description(), squeezeAxes, gatherOutRank);
+                std::sort(squeezeAxes.begin(), squeezeAxes.end(), [](int64_t a, int64_t b) { return a > b; });
```

Comment:
Nitpick: is it possible to use `std::greater`?

Reply:
Why do you need sort? Do I understand correctly that you have squeezeOutIndices as a vector of elements from which you want remove those which are specified in squeezeAxes?

Reply:
Yes, you're right, I got rid of this sort, please take another look.